yogstation/code/modules/antagonists/darkspawn/darkspawn_abilities/_divulge.dm

```diff
@@ -1,91 +1,123 @@
 //A channeled ability that turns the darkspawn into their main form.
-/datum/action/innate/darkspawn/divulge
+/datum/action/cooldown/spell/divulge
 	name = "Divulge"
-	id = "divulge"
 	desc = "Sheds your human disguise. This is obvious and so should be done in a secluded area. You cannot reverse this."
+	panel = "Darkspawn"
+	button_icon = 'yogstation/icons/mob/actions/actions_darkspawn.dmi'
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	buttontooltipstyle = "alien"
 	button_icon_state = "divulge"
 	check_flags =  AB_CHECK_IMMOBILE | AB_CHECK_CONSCIOUS | AB_CHECK_LYING
-	blacklisted = TRUE
+	spell_requirements = NONE
+	///if they're actively divulging
+	var/in_use = FALSE
 
-/datum/action/innate/darkspawn/divulge/Activate()
+/datum/action/cooldown/spell/divulge/IsAvailable(feedback)
+	if(in_use)
+		if (feedback)
+			owner.balloon_alert(owner, "already in use!")
+		return
+	return ..()
+
+/datum/action/cooldown/spell/divulge/cast(atom/cast_on)
 	set waitfor = FALSE
+	. = ..()
 	var/mob/living/carbon/human/user = usr
 	var/turf/spot = get_turf(user)
 	if(!ishuman(user))
 		to_chat(user, span_warning("You need to be human-er to do that!"))
 		return
-	if(isethereal(user))
-		user.set_light(0)
-	if(spot.get_lumcount() > DARKSPAWN_DIM_LIGHT)
+
+	if(isdarkspawn(user))
+		var/datum/antagonist/darkspawn/darkspawn = isdarkspawn(user)
+		if(!darkspawn.picked_class)
+			to_chat(user, span_warning("You must pick a class to divulge!"))
+			return
+
+	if(isethereal(user))//disable the light for long enough to start divulge
+		user.dna.species.spec_emp_act(user, EMP_HEAVY)
+			
+	if(spot.get_lumcount() > SHADOW_SPECIES_DIM_LIGHT)
 		to_chat(user, span_warning("You are only able to divulge in darkness!"))
 		return
-	if(alert(user, "You are ready to divulge. Are you sure?", name, "Yes", "No") == "No")
+
+	if(alert(user, "You are ready to divulge. This will leave you vulnerable and take a long time, are you sure?", name, "Yes", "No") == "No")
```

Comment:
Can this use tgui alert instead?
You'd also have to swap this to do ``!= "Yes"`` instead cause otherwise x'ing out of the alert is taken as a Yes.